backend/src/lib/services/s3.ts

```diff
@@ -19,6 +20,17 @@ class S3Service {
     return S3Service.instance;
   }
 
+  async putDataset(bucketName: string, jsonS3Key: string, jsonFile: string) {
```

Comment:
I think the `S3Service` shouldn't have knowledge of the concept of `Dataset`. I would suggest to rename this function to `putObject` instead so that it is generic for any type of file.